### PR TITLE
[OTHER]: Pull Heroic latest fixes and features

### PR DESCRIPTION
**Pull request by flavioislima**
Got some fixes that Heroic had recently.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by flavioislima** [comment]
@stormcakes There are a few things to test on this one:
- On the Add Game Dialog, there is a bug on the current release that the Finish button is not clickable if an image for the game was not found. This should be fixed on this PR.
- On the Epic Login page, the captcha was not working and was always returning an error with an invalid response. Should be fixed as well.
- Sideloaded games (added on the Add Game dialog), did not have the play status working correctly. They changed to Playing but never got out of it even after the game got close.
- And the latest thing to test is the new feature of searching for a game on the search bar and clicking on it should take you to the game page.

Other than that are minor changes that do not need to be deeply tested.

**Comment by flavioislima** [comment]
Added one last fix to the Add Game dialog:
- On the current release all platforms are showing on all OSes, that is wrong since you cannot run a macOS app on linux and vice-versa. This should be fixed now.

**Comment by stormcakes** [comment]
On windows OS when sideloading a game (I used League of Legends) and playing the game, while the game is open the Play button and "Playing" text goes away and the user can click "Play" again. @flavioislima

**Comment by stormcakes** [comment]
Verified the following so far:

- On the Add Game Dialog, there is a bug on the current release that the Finish button is not clickable if an image for the game was not found. This should be fixed on this PR.
- And the latest thing to test is the new feature of searching for a game on the search bar and clicking on it should take you to the game page.

Issues found with: 
-Sideloaded games (added on the Add Game dialog), did not have the play status working correctly. They changed to Playing but never got out of it even after the game got close. [I can create a separate bug if needed.]

WIP:
- On the Epic Login page, the captcha was not working and was always returning an error with an invalid response. Should be fixed as well.
- On the current release all platforms are showing on all OSes, that is wrong since you cannot run a macOS app on linux and vice-versa. This should be fixed now.

**Comment by stormcakes** [comment]
All testing so far has been performed on Windows - will expand to MacOS (both M1 and Intel) soon and then move to Linux.

**Review by BrettCleary — CHANGES_REQUESTED** [review]

**Comment by BrettCleary** [comment]
> Sideloaded games (added on the Add Game dialog), did not have the play status working correctly. They changed to Playing but never got out of it even after the game got close. [I can create a separate bug if needed.]

This is a separate issue. Basically sometimes when you sideload a game you launch an exe that launches another exe but the play status/stop button only affects/tracks the first exe

**Comment by BrettCleary** [comment]
> On the Add Game Dialog, there is a bug on the current release that the Finish button is not clickable if an image for the game was not found. This should be fixed on this PR.
> On the Epic Login page, the captcha was not working and was always returning an error with an invalid response. Should be fixed as well.
> Sideloaded games (added on the Add Game dialog), did not have the play status working correctly. They changed to Playing but never got out of it even after the game got close.
> And the latest thing to test is the new feature of searching for a game on the search bar and clicking on it should take you to the game page.

Tested and these are all working

**Comment by stormcakes** [comment]
I can make a seperate ticket for the bug of sideloading but everything else looks good to me. I didnt run into the captcha verification but i was able to auth :)

**Review by stormcakes — APPROVED** [review]
From the test pov besides the sideloading bug 
(which will be written up seperately) - everything is working fine so i'll go ahead and approve :)

File: src/backend/storeManagers/sideload/games.ts [hunk 1]
```diff
@@ -82,6 +82,11 @@ export async function stop(appName: string): Promise<void> {
     const split = executable.split('/')
     const exe = split[split.length - 1]
     killPattern(exe)
+
+    if (!isNative(appName)) {
+      const gameSettings = await getSettings(appName)
+      shutdownWine(gameSettings)
+    }
```
[the remark this answers is not in the document]
Comment (flavioislima) [reply src/backend/storeManagers/sideload/games.ts:89]: done

File: src/backend/storeManagers/sideload/library.ts [hunk 2]
```diff
@@ -15,8 +15,8 @@ export function addNewApp({
   app_name,
   title,
   install: { executable, platform },
-  art_cover,
-  art_square,
+  art_cover = 'fallback',
+  art_square = 'fallback',
```
[the remark this answers is not in the document]
Comment (flavioislima) [reply src/backend/storeManagers/sideload/library.ts:19]: done

File: src/frontend/helpers/gamepad.ts [hunk 3]
```diff
@@ -334,26 +334,32 @@ export const initGamepad = () => {
...
+      if (axes[axis] < -0.2 && axes[axis] >= -1)
+        console.log(`axis ${axis} activated negative`)
+      if (axes[axis] > 0.2 && axes[axis] <= 1)
+        console.log(`axis ${axis} activated positive`)
+    }
+  }
 
   function connecthandler(e: GamepadEvent) {
+    console.log('controller connected event')
+    console.log(e)
```
[the remark this answers is not in the document]
Comment (flavioislima) [reply src/frontend/helpers/gamepad.ts]: yes, done

**Review by BrettCleary — APPROVED** [review]
LGTM

**Pull request merged**